src/librustdoc/passes/unindent_comments.rs

```diff
@@ -35,65 +34,72 @@ impl clean::Attributes {
-            saw_first_line && !saw_second_line && !line.chars().all(|c| c.is_whitespace());
 
-        let min_indent = if ignore_previous_indents { usize::MAX } else { min_indent };
+    let mut min_indents = docs.iter().map(|fragment| {
+        fragment.doc.lines().fold(usize::MAX, |min_indent, line| {
+            // After we see the first non-whitespace line, look at
+            // the line we have. If it is not whitespace, and therefore
+            // part of the first paragraph, then ignore the indentation
+            // level of the first line
+            let ignore_previous_indents =
+                saw_first_line && !saw_second_line && !line.chars().all(|c| c.is_whitespace());
 
-        if saw_first_line {
-            saw_second_line = true;
-        }
+            let min_indent = if ignore_previous_indents { usize::MAX } else { min_indent };
 
-        if line.chars().all(|c| c.is_whitespace()) {
-            min_indent
-        } else {
-            saw_first_line = true;
-            let mut whitespace = 0;
-            line.chars().all(|char| {
-                // Compare against either space or tab, ignoring whether they
-                // are mixed or not
-                if char == ' ' || char == '\t' {
-                    whitespace += 1;
-                    true
-                } else {
-                    false
-                }
-            });
-            cmp::min(min_indent, whitespace)
-        }
-    });
+            if saw_first_line {
+                saw_second_line = true;
+            }
 
-    if !lines.is_empty() {
-        let mut unindented = vec![lines[0].trim_start().to_string()];
-        unindented.extend_from_slice(
-            &lines[1..]
-                .iter()
-                .map(|&line| {
-                    if line.chars().all(|c| c.is_whitespace()) {
-                        line.to_string()
+            if line.chars().all(|c| c.is_whitespace()) {
+                min_indent
+            } else {
+                saw_first_line = true;
+                let mut whitespace = 0;
+                line.chars().all(|char| {
+                    // Compare against either space or tab, ignoring whether they
+                    // are mixed or not
+                    if char == ' ' || char == '\t' {
+                        whitespace += 1;
+                        true
                     } else {
-                        assert!(line.len() >= min_indent);
-                        line[min_indent..].to_string()
+                        false
                     }
-                })
-                .collect::<Vec<_>>(),
-        );
-        unindented.join("\n")
-    } else {
-        s.to_string()
+                });
+                cmp::min(min_indent, whitespace)
+            }
+        })
+    }).collect::<Vec<_>>();
+    if min_indents.is_empty() {
+        return;
+    }
+    min_indents.sort_unstable();
+    let min_indent = min_indents[0];
```

Comment:
Why did all this have to change? From what I see you don't use `min_indents` after this.

Reply:
I needed to get the `min_indent` based on all the doc fragments, not just on one.